packages/fxa-auth-server/lib/routes/oauth/index.js

```diff
@@ -79,6 +81,36 @@ module.exports = (log, config, oauthdb, db, mailer, devices) => {
         return oauthdb.getScopedKeyData(sessionToken, request.payload);
       },
     },
+    // TODO moving this API directly into auth server here, instead of
+    // bothering with the oauth-server indirection.
+    {
+      method: 'POST',
+      path: '/oauth/id-token-verify',
+      options: {
+        auth: {
+          strategy: 'sessionToken',
+        },
+        validate: {
+          headers: clientAuthValidators.headers,
```

Comment:
This adds support for parsing `client_id` and `client_secret` out of the `Authorization` header, but I don't think you want that here. IIUC we don't intend for clients to call this endpoint directly anyway, and you're not doing anything that the caller couldn't do for itself. So, no need for secrets here.

Reply:
Ah, I was thinking requiring a sessionToken might make it a little harder for an attacker to experiment with our validator, but happy to remove it :+1:

Reply:
Hrm, interesting. I don't think it would add any additional security, but having this be authenticated by a sessionToken might help us just keep the whole thing on the rails a little more effectively - e.g. it could make it harder to accidentally accept an `id_token` when the user has an invalid sessionToken, or make it harder to accidentally mix up an `id_token` for user `A` with a sessionToken for user `B`. I would be happy to see that explored if it makes sense to you.

(You still wouldn't want the `clientAuthValidators` header here, but rather a sessionToken auth stragegy)